model/dns_zone/dns_zone.rb

```diff
@@ -52,6 +52,40 @@ def delete_record(record_name:, type: nil, data: nil)
     incr_refresh_dns_servers
   end
 
+  def purge
+    DB.transaction do
```

Comment:
State machines already run in transaction. You don't need transaction block here.

Reply:
Since it is in the model, we can run it through the model in `pry`. Therefore, having DB.transaction could be beneficial.

Reply:
Exactly my thoughts

Reply:
Makes sense.